src/IO/WriteBufferFromS3.cpp

```diff
@@ -703,6 +705,12 @@ S3::PutObjectRequest WriteBufferFromS3::getPutRequest(PartData & data)
     req.SetKey(key);
     req.SetContentLength(data.data_size);
     req.SetBody(data.createAwsBuffer());
+
+    // By default AWS SDK uses MD5 as a checksum algorithm.
+    // MD5 is not supported in OpenSSL FIPS mode as it is considered weak and collision-prone.
+    // This has to be a checksum supported by AWS S3, GCS and MiniIO.
+    req.SetChecksumAlgorithm(Aws::S3::Model::ChecksumAlgorithm::CRC32C);
```

Comment:
Looking at https://en.wikipedia.org/wiki/Cyclic_redundancy_check , CRC32 seems FIPS compliant but CRC32C not. Would you double-check?

(let's also include a link to some source which proves that the checksum algorithm is FIPS compliant).

Reply:
:raised_eyebrow: 
https://patchwork.kernel.org/project/linux-crypto/patch/20190125093147.10266-1-gmazyland@gmail.com/

Reply:
I am also no expert in FIPS compliance :-) and of course, S3 doesn't use checksumming for cryptographic purposes. But I would say better be safe than sorry - we should minimize the chance that auditors complain later on.

I also found

https://stackoverflow.com/questions/14130110/hashing-algorithm-that-meet-fisma-other-federal-informations-systems-requireme

which suggests SHA-512 for FIPS complaince (and SHA-512 is also available in the AWS SDK).

Anyways, perhaps standard CRC32 is good. Changing it afterwards would be a minor change.